fedimint-core/src/amount.rs

```diff
@@ -211,13 +211,13 @@ impl FromStr for Amount {
             Self::from_str_in(amt.trim(), denom.trim().parse()?)
```

Comment:
Note for reviewer: We added a `WrongBitcoinDenomination` error type below because `denom.trim().parse()` now returns a `ParseDenominationError` instead of a `ParseAmountError`:

v0.30: https://docs.rs/bitcoin/0.30.2/src/bitcoin/amount.rs.html#93
v0.32: https://docs.rs/bitcoin-units/0.1.2/src/bitcoin_units/amount.rs.html#126

(Also, we do still need `ParseAmountError::WrongBitcoinAmount` since above we call `Amount::from_str_in(s, denom)?`)